libdevice/imf_bf16.hpp

```diff
@@ -81,6 +81,85 @@ __float2bfloat16(float f, __iml_rounding_mode rounding_mode) {
 
   return (bf16_sign << 15) | (bf16_exp << 7) | bf16_mant;
 }
+// We only need utils to convert double to bfloat16 with RTE
+static _iml_bf16_internal __double2bfloat16(double d) {
+  union {
+    double d_val;
+    uint64_t u64_val;
+  } fp64_bits;
+
+  fp64_bits.d_val = d;
+  uint16_t bf16_sign =
+      static_cast<uint16_t>((fp64_bits.u64_val & 0x8000000000000000) >> 63);
```

Comment:
let's not pun through union.
i think that__builtin_bitcast is for that.

I believe 
(val >> 63) & 0x1
(val >> 52) & 0x7FF
is better as it avoids long constants.
But it is a matter of taste.

Reply:
Done.
Thanks very much.